<launch>
  <arg name="rgb_image_topic" default="/camera/rgb/image_rect_color" />
  <arg name="depth_image_topic" default="/camera/depth_registered/image_raw" />
  <arg name="rgb_camera_info_topic" default="/camera/rgb/camera_info" />

  <!-- For rosbag without recorded pointcloud -->
  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

  <node pkg="nodelet" type="nodelet" name="nodelet1" args="load depth_image_proc/point_cloud_xyz nodelet_manager">
    <remap from="camera_info" to="/camera/depth_registered/camera_info" />
    <remap from="image_rect" to="/camera/depth_registered/image_raw" />
    <remap from="points" to="/camera/depth_registered/points" />
  </node>

  <!-- Kinect cloud to laser scan -->
  <!-- consider using nodelet instead -->
  <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan_node">
    <remap from="cloud_in" to="/camera/depth_registered/points" />
    <remap from="scan" to="/kinect_scan" />
    <param name="min_height" type="double" value="0.0" />
    <param name="max_height" type="double" value="1.0" />
    <param name="target_frame" type="string" value="base_link" />
    <param name="angle_increment" type="double" value="0.00174532923847" />


  </node>

  <!-- SLAM -->
  <group ns="rtabmap">
    <node pkg="nodelet" type="nodelet" name="rgbd_sync" args="standalone rtabmap_ros/rgbd_sync" output="screen">
      <remap from="rgb/image" to="$(arg rgb_image_topic)" />
      <remap from="depth/image" to="$(arg depth_image_topic)" />
      <remap from="rgb/camera_info" to="$(arg rgb_camera_info_topic)" />
      <remap from="rgbd_image" to="rgbd_image" /> <!-- output -->

      <!-- Should be true for not synchronized camera topics 
           (e.g., false for kinectv2, zed, realsense, true for xtion, kinect360)-->
      <param name="approx_sync" value="true" />
    </node>

    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
      <remap from="odom" to="/odom" />
      <remap from="scan" to="/kinect_scan" />

      <remap from="rgb/image" to="/camera/rgb/image_rect_color" />
      <remap from="depth/image" to="/camera/depth_registered/image_raw" />
      <remap from="rgb/camera_info" to="/camera/rgb/camera_info" />

      <rosparam file="$(find roomac_rtabmap)/dev_utils/alternative_rtabmap_configs/rtabmap_laserscan_wheel_odom.yaml" command="load" />
    </node>
  </group>
</launch>